sol/contracts/ProtoBufRuntime.sol

```diff
@@ -0,0 +1,2990 @@
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
The biggest issue here is the use of this library. It is a protobuf codec that may or may not follow a non-existent protobuf spec, not an [ADR-027](https://github.com/cosmos/cosmos-sdk/blob/b0d3ef96506054b66bc1160d48f4b24105e7d35a/docs/architecture/adr-027-deterministic-protobuf-serialization.md) codec. The latter is highly recommended (read: required) to avoid the possibility of malleability attacks such as [this one](https://medium.com/immunefi/polygon-double-spend-bug-fix-postmortem-2m-bounty-5a1db09db7f1). You should be using the following libraries to deal with protobuf.

1. https://github.com/celestiaorg/protobuf3-solidity
2. https://github.com/celestiaorg/protobuf3-solidity-lib

Reply:
HI @adlerjohn ,
the problem with the ``celestiaorg`` compiler is that, as reported in the README, it does not support ``oneof``  needed in [``BatchEntry``](https://github.com/confio/ics23/blob/e736db6536ced19ed677b4d4d00ade63d47d4a3d/proofs.proto#L199)

Reply:
`oneof` can be implemented as a library! Much better than fixing the currently-used library to be ADR-027-compliant. PRs welcome.

Reply:
I have talked to the [datachainlab/solidity-protobuf](https://github.com/datachainlab/solidity-protobuf) compiler's maintainers.
They have plans to make it ADR-27 compliant soon.
I think I will just sit and wait for it :grin: